solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -177,27 +191,42 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
         return amount;
     }
 
-    function _write() internal {
-        uint256 price0 =
-            ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
-                reserveBalance(reserveTokens[0]),
-                reserves[reserveTokens[0]].weight,
-                reserveBalance(reserveTokens[1]),
-                reserves[reserveTokens[1]].weight,
-                1
-            );
-
-        uint256 price1 =
-            ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
-                reserveBalance(reserveTokens[1]),
-                reserves[reserveTokens[1]].weight,
-                reserveBalance(reserveTokens[0]),
-                reserves[reserveTokens[0]].weight,
-                1
-            );
-
-        oracle.write(price0, price1);
-    }
+	function _write() internal {
+        IERC20Token token0 = IERC20Token(reserveTokens[0]);
+        IERC20Token token1 = IERC20Token(reserveTokens[1]);
+
+        uint256 reserve0Balance = reserveBalance(token0);
+        uint256 reserve1Balance = reserveBalance(token1);
+
+        uint32 reserve0Weight = reserves[token0].weight;
+        uint32 reserve1Weight = reserves[token1].weight;
+
+        uint256 partialReserves0 = reserve0Balance.div(DENOMINATOR);
+        uint256 partialReserves1 = reserve1Balance.div(DENOMINATOR);
+
+        ISovrynSwapFormula sovrynSwapFormula = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)); 
+
+		uint256 price0 = sovrynSwapFormula.crossReserveTargetAmount(
+			reserve0Balance,
+			reserve0Weight,
+			reserve1Balance,
+			reserve1Weight,
+			partialReserves0
+		);
+
+		uint256 price1 = sovrynSwapFormula.crossReserveTargetAmount(
+			reserve1Balance,
+			reserve1Weight,
+			reserve0Balance,
+			reserve0Weight,
+			partialReserves1
+		);
+		
+		price0 = price0.mul(10**token0Decimal).div(partialReserves0);
+		price1 = price1.mul(10**token1Decimal).div(partialReserves1);

```

Comment:
Are you sure about this calculation specifically that you are dividing by this value? If I understand correctly you want to calculate the price for 1 source token. But then this is not entirely correct since they are not proportional

Reply:
The value ratio between the tokens of a v1 pool stays always the same. In our case, we're always using 50/50. The exchange rate is determined by the current reserve balances assuming that the value of the reserve balances is always the same (because 50/50).  

Here, we return the value of 1 token. One option would be to directly query the rate of 1 token (including decimals). However, this would mean that we have very different slippage considered in the price for both directions.

Example: 1 MoC is worth 0.00000616252 BTC. In the pool there are approx. 2.5 BTC and 382000 MOC. Querying the rate for 1 BTC will include massive slippage, while 1 MoC has close to 0 slippage.

This is why Rahne is here querying the rate of 0.1% of the repsective reserve balance, so the slippage is the same in both directions, and then calculating the rate per token based on that number. 

Is there an issue I'm not able to see?